rag-vector-sveltekit-xata.mdx

```diff
@@ -0,0 +1,397 @@
+
+```typescript
+// Fetch the relevant set of records based on the embedding
+const relevantRecords = await xata.db.vectors.vectorSearch(
+  // Column Name
+  'embedding',
+  // Embedding Vector
+  embeddingData.data[0].embedding,
+  // Count of relevant vector embeddings
+  { size: 5 }
+);
+```
+
+With each relevant embedding vector, each record of your Xata database is returned. This allows us to obtain the metadata associated with these vectors and use them to create the context for the AI to answer with. In the code below, we simply create a string containing all the metadata from the relevant vectors returned in our search.
+
+```typescript
+// File: src/routes/api/context/warm/+server.ts
+
+import { embedding } from 'litellm';
+import { getXataClient } from '@/xata.server';
+
+// Load the in-memory Xata client
+const xata = getXataClient();
+
+export async function POST({ request }: RequestEvent) {
+  // Set of messages to create vector embeddings on
+  const { messages = [] } = await request.json();
+
+  // Get the latest question stored in the last message of the chat array
+  const userMessages = messages.filter((i: any) => i.role === 'user');
+  const input = userMessages[userMessages.length - 1].content;
+
+  // Generate embeddings of the latest question using OpenAI
+  const embeddingData = await embedding({ input, model: 'text-embedding-3-small' });
+
+  // Fetch the relevant set of records based on the embedding
+  const relevantRecords = await xata.db.vectors.vectorSearch('embedding', embeddingData.data[0].embedding, { size: 5 });
+
+  // Combine all the metadata of the relevant vectors
+  const systemContext = relevantRecords.records.map((i) => i.contents).join('\n');
+}
+```
+
+## Using Replicate Predictions to Prompt LLAMA 2 70B Chat Model
+
+To prompt Meta’s LLAMA 2 70B Chat model, we’ll use Replicate which allows us to run the model with just an API call. In our case, we make use of the `replicate` package, saving us from the need to write a fetch api by ourselves and handle streaming responses natively. We need to make sure that the `REPLICATE_API_TOKEN` is set as an environment variable. Go to your [Replicate’s Account Settings > API tokens](https://replicate.com/account/api-tokens) to obtain your Replicate API token.
+
+```typescript
+import Replicate from 'replicate'
+
+// Instantiate the Replicate API
+const replicate = new Replicate({
+  auth: process.env.REPLICATE_API_TOKEN,
+}
+```
+
+Using Vercel’s AI SDK baked-in `experimental_buildLlama2Prompt` function, we’re able to:
+
+1.  Pass instructions so that the code blocks in the markdown-like responses generated by AI are prepended with \`\`\`\`\`. This'll help us highlight them in our frontend setup via a syntax highlighter. This is done via creating a message with the role of `assistant`and passing our relevant instructions in the`content` field.
```

Comment:
```suggestion
1.  Pass instructions so that the code blocks in the markdown-like responses generated by AI are prepended with \`\`\`. This'll help us highlight them in our frontend setup via a syntax highlighter. This is done via creating a message with the role of `assistant`and passing our relevant instructions in the`content` field.
```